### PR TITLE
[Search/RoadMap] Some improvements & bug fix

### PR DESCRIPTION
**Pull request by indydedeken**
- «RoadMap.addMarkerSearch» is now «RoadMap.setRequestedPositionMarker».
- Fix double marker when multiples places are requested (from
  setRequestedPositionMarker())
- addMarkers very little improvement by splitting the markers add.
- Geolocation is now waiting for stationStorage is loaded in order to
  setPositionMarker on the roadMap.

### CODE REVIEW

**Comment by eliemichel** [comment]
Why do you need `stationStorage` to add a `positionMarker` on the `roadMap`?

File: js/Control.OSMGeocoder.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ L.Control.OSMGeocoder = L.Control.extend({
 			this._map.fitBounds(bounds);
 
 			// IDE - Map have the power to draw a marker
-			RoadMap.addMarkerSearch(this._map.fitBounds(bounds));
+			RoadMap.setRequestedPositionMarker(this._map.fitBounds(bounds));
```
Comment (eliemichel) [inline js/Control.OSMGeocoder.js:22]: You should not modify vendor libraries as they could be updated.
Actually, as soon as you modify a library, you have to care about its updates to merge your modification at any time, i.e. you have to maintain it. So we better avoid it!

**Comment by indydedeken** [comment]
I missed it, bad copy-paste. I will delete it.

File: js/Control.OSMGeocoder.js (hunk 1 above)
Comment (indydedeken) [inline js/Control.OSMGeocoder.js:22]: Damage is already done. Here it's a simple function renaming.

Can I open an issue an fix it later ? Vendor code is not so much impacted.

Comment (Phyks) [inline js/Control.OSMGeocoder.js:22]: If you simply want to rename the function, you can also define an alias in the code :)

As I think this vendor code was already modified in the past, I think it is reasonable to open an issue and revert all the vendor code later.

Comment (eliemichel) [inline js/Control.OSMGeocoder.js:22]: Yep, I will open a new issue.

File: js/RoadMap.js [hunk 2]
```diff
@@ -62,47 +63,53 @@ var RoadMap = (function() {
...
+    var addMarkers = (function(_stationStorage) {
+        console.log('RoadMap', 'addMarkers()');
+        stationStorage = _stationStorage;
+
         var myIcon = '';
+        var markersList = [];
 
-        for(var i=0; i < stations.length; i++) {
+        // build Markers
+        for(var i=0; i < stationStorage.length; i++) {
```
Comment (eliemichel) [inline js/RoadMap.js:74]: Does `stationStorage` really have a `length` attribute?

Comment (eliemichel) [inline js/RoadMap.js:74]: Oh, there is a `stationStorage = stationStorage.getStations()`.
Please avoid reusing the same variable name when data type changes.
In the right-hand member, `stationStorage` is a `StationStorage` (this is why it is called like so). But in the left-hand member, it is a list of stations.

It is really confusing actually, and `stationStorage` is not a good name for a list of stations, as it is already associated with another type of object. You could name it `stations` or `stationList` for instance.

Furthermore, it breaks a global variable (within the Black Box scope) which could be used in some other places as the original instance of `StationStorage`. This is an unexpected side effect.

**Pull request closed without merging**